openlattice/lattice-fabricate
# Release 12-19-2019

# Changes
- Add `Paged` component to help create multi-paged `Form` components.
  - Tested!
- BaseInput is now passed the correct `type` to help mobile keyboards when schema `type: 'number'`
- Extend SelectWidget
  - receives `onBlur` and `onFocus` from the `Form`
  - expose `multiple` and `noOptionsMessage` ui:options
  - `blurInputOnSelect` and `closeMenuOnSelect` are set to false when `multiple` is true
  - Before, mobile users could not create multiple elements by pressing enter. Now, with custom `noOptionsMessage`, `hideMenu` is now deprecated and should be avoided to better support mobile users.
- Add DateTimeWidget

```
// schema
datetime: { 
  type: 'string',
  format: 'date-time'
  title: 'Date & Time'
}

// uischema
{
  'ui:widget': 'DateTimeWidget' // not necessary to set if format is set to 'date-time'
}
```
- Add CheckboxesWidget
```
// schema
checkboxes: {
  title: 'Options',
  type: 'array',
  items: {
    type: 'string',
    enum: ['Option 1', 'Option 2', 'Option 3', 'Other'],
  },
  minItems: 1,
  uniqueItems: true,
}
// uischema
{
  checkboxes: {
    classNames: 'column-span-12',
    'ui:widget': 'checkboxes', // or 'CheckboxesWidget'
    'ui:options': {
      withOther: true // if you want an input to popup when `Other` is selected
    }
  }
}
```

# Fixes
- Error messages are now keyed properly
